File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;
        HashMap<Integer, Integer> map = new HashMap<>();
        // Taking the Element into the Hashmap
        for (int i = 0; i < n; i++) {
            map.put(nums[i], i);
        }

        // Checking the result
        for (int i = 0; i < n; i++) {
            int number = nums[i];
            int rem = target - number;
            if (map.containsKey(rem)) {
                int idx = map.get(rem);
                // Condition for same index
                if (idx == i)
                    continue;
                return new int[] { i, idx };
            }
        }
        return new int[] {};
    }
}

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const map = new Map(); // {key: element, value: index}
    
    // Mapping the elements with it's index:
    for(let i=0; i<nums.length; i++){
        map.set(nums[i], i);
    }
    
    // Checking for the Complement:
    for(let i=0; i<nums.length; i++){
        let complement = target - nums[i];
        
        // Checking into the Map:
        if(map.has(complement) && map.get(complement) != i)
            return [i, map.get(complement)];
    }
};